qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -444,10 +444,308 @@ def particle_number(orbitals, mapping="jordan_wigner", wires=None):
+                    orbitals, core
+                )
+            )
+
+        # Compute contribution due to core orbitals
+        t_core = 2 * sum([t_matrix_elements[alpha, alpha] for alpha in core])
+
+    if active is None:
+        if core is None:
+            active = list(range(orbitals))
+        else:
+            active = [i for i in range(orbitals) if i not in core]
+
+    if True in [i > orbitals - 1 or i < 0 for i in active]:
+        raise ValueError(
+            "Indices of active orbitals must be between 0 and {}; got active = {}".format(
+                orbitals, active
+            )
+        )
+
+    # Indices of the matrix elements with absolute values >= cutoff
+    indices = np.nonzero(np.abs(t_matrix_elements) >= cutoff)
+
+    # Single out the indices of active orbitals
+    num_indices = len(indices[0])
+    pairs = [
+        [indices[0][i], indices[1][i]]
+        for i in range(num_indices)
+        if all(indices[j][i] in active for j in range(len(indices)))
+    ]
+
+    # Building the table of indices and matrix elements
+    t_table = np.zeros((2 * len(pairs), 3))
+
+    for i, pair in enumerate(pairs):
+        alpha, beta = pair
+        element = t_matrix_elements[alpha, beta]
+
+        # spin-up term
+        t_table[2 * i, 0] = 2 * active.index(alpha)
+        t_table[2 * i, 1] = 2 * active.index(beta)
+        t_table[2 * i, 2] = element
+
+        # spin-down term
+        t_table[2 * i + 1, 0] = 2 * active.index(alpha) + 1
+        t_table[2 * i + 1, 1] = 2 * active.index(beta) + 1
+        t_table[2 * i + 1, 2] = element
+
+    return t_table, t_core
+
+
+def two_particle(v_matrix_elements, core=None, active=None, cutoff=1.0e-12):
+    r"""Generates the table of matrix elements of a given two-particle operator
+    required to build many-body qubit observables.
+
+    Second quantized two-particle operators are expanded in the basis of single-particle
+    states as
+
+    .. math::
+
+        \hat{V} = \frac{1}{2} \sum_{\alpha, \beta, \gamma, \delta}
+        \langle \alpha, \beta \vert \hat{v} \vert \gamma, \delta \rangle
+        ~ &[& \hat{c}_{\alpha\uparrow}^\dagger \hat{c}_{\beta\uparrow}^\dagger
+        \hat{c}_{\gamma\uparrow} \hat{c}_{\delta\uparrow} + \hat{c}_{\alpha\uparrow}^\dagger
+        \hat{c}_{\beta\downarrow}^\dagger \hat{c}_{\gamma\downarrow} \hat{c}_{\delta\uparrow} \\
+        &+& \hat{c}_{\alpha\downarrow}^\dagger \hat{c}_{\beta\uparrow}^\dagger
+        \hat{c}_{\gamma\uparrow} \hat{c}_{\delta\downarrow} + \hat{c}_{\alpha\downarrow}^\dagger
+        \hat{c}_{\beta\downarrow}^\dagger \hat{c}_{\gamma\downarrow} \hat{c}_{\delta\downarrow}~].
+
+    In the equation above the indices :math:`\alpha, \beta, \gamma, \delta` run over the basis
+    of spatial orbitals :math:`\vert \alpha \rangle = \phi_\alpha(r)`. Since the operator
+    :math:`v` acts only on the spatial the spin quantum numbers are indicated explicitly
```

Comment:
This sentence is missing a part, it now reads "acts only on the spatial the spin quantum...". I guess it should be "acts only on the spatial orbitals, the spin quantum..."

Reply:
Good catch!, thanks @ixfoduap